modules/http/src/test/java/org/mule/module/http/functional/requester/ocsp/AbstractHttpOcspRevocationTestCase.java

```diff
@@ -0,0 +1,105 @@
+ */
+
+package org.mule.module.http.functional.requester.ocsp;
+
+import static java.lang.String.format;
+import static org.junit.Assume.assumeFalse;
+import org.mule.module.http.functional.AbstractHttpTlsRevocationTestCase;
+import org.mule.tck.junit4.rule.DynamicPort;
+import org.mule.tck.junit4.rule.SystemProperty;
+
+import java.io.IOException;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+
+public abstract class AbstractHttpOcspRevocationTestCase extends AbstractHttpTlsRevocationTestCase
+{
+
+    @Rule
+    public DynamicPort dynamicOcspPort ;
+
+    @Rule
+    public SystemProperty certAliasSystemProperty ;
+
+    private static String RUN_OCSP_SERVER_COMMAND = "openssl ocsp " +
+                                                    "-index src/test/resources/tls/ocsp/%s " +
+                                                    "-port %d " +
+                                                    "-CA src/test/resources/tls/ocsp/server.crt " +
+                                                    "-rkey src/test/resources/tls/ocsp/%s.pem " +
+                                                    "-rsigner src/test/resources/tls/ocsp/%s.crt";
+
+    static final String REVOKED_OCSP_LIST = "revoked-ocsp.txt";
+
+    static final String VALID_OCSP_LIST = "valid-ocsp.txt";
+
+    /**
+     *  This certified entity was generated to test revocation with OCSP mechanism.
+     */
+    protected static final String ENTITY_CERTIFIED_REVOCATION_OCSP_SUB_PATH = "entityOcsp1";
+
+    /**
+     * The entity4 certificate has a hardcoded ocsp url: http://localhost:1111
+     */
+    static final int FIXED_OCSP_PORT = 1111;
+
+
+    static final String DEFAULT_OCSP_RESPONDER= "server";
+
+    private Process process ;
+
+    private final String ocspList ;
+
+    private int ocspPort;
+
+    private String ocspResponder ;
+
+
+    AbstractHttpOcspRevocationTestCase(String configFile, String ocspList, String certAlias)
+    {
+        super(configFile, ENTITY_CERTIFIED_REVOCATION_OCSP_SUB_PATH);
+        this.ocspList = ocspList;
+        initialiseOcspStatus(certAlias);
+    }
+
+    private void initialiseOcspStatus (String certAlias)
+    {
+        if (certAlias != null)
+        {
+            certAliasSystemProperty = new SystemProperty("certAlias", certAlias);
+            dynamicOcspPort = new DynamicPort("dynamicOcspPort");
+            ocspPort = dynamicOcspPort.getNumber();
+            ocspResponder = certAliasSystemProperty.getValue();
+        }
+        else
+        {
+            ocspPort = FIXED_OCSP_PORT ;
+            ocspResponder = DEFAULT_OCSP_RESPONDER;
+        }
+    }
+
+
+    @Before
+    public void setUp() throws Exception
+    {
+        // Running the ocsp server in a fixed port causes that sometimes the server fails at initialisation.
+        assumeFalse("Since openssl ocsp command has a flaky behavior the test will be ignored when the port number is not dynamic.", FIXED_OCSP_PORT == ocspPort);
```

Comment:
Does this mean this won't run in parallel with other tests that might get 1111 assigned through `DynamicPort`?

Reply:
Looks like since you're trying to validate that the dynamic port is the one used if 1111 is assigned the test won't work, however, the reason mentions openssl flakiness. Could you clarify this?

Reply:
The openssl command has an issue reopening the same port one closed. If you run the command, send an OCSP query, close the process (even wait a few seconds) and then start it again in the same port it fails.
It seems to be trying to reuse it...

Reply:
We can't change the port in the certificate because it's generated manually.